### PR TITLE
Adding bases for molecule tests

### PR DESCRIPTION
**Pull request by cjeanner**
The whole was mostly taken from the edpm-ansible[1] project, since they have the same kind of requirements.

[1] https://github.com/openstack-k8s-operators/edpm-ansible/pull/33/files

### CODE REVIEW

File: bindep.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+libffi-devel      [platform:rpm]
+openssl-devel     [platform:rpm]
+podman            [platform:rpm]
+python3-devel     [platform:rpm !platform:rhel-7 !platform:centos-7]
+PyYAML            [platform:rpm !platform:rhel-8 !platform:centos-8 !platform:rhel-9 !platform:centos-9 !platform:fedora]
+python3-pyyaml    [platform:rpm !platform:rhel-7 !platform:centos-7]
+python3-dnf       [platform:rpm !platform:rhel-7 !platform:centos-7]
+
+# RH Mechanisms
+python-rhsm-certificates                [platform:redhat]
```
Comment (raukadah) [inline bindep.txt:28]: please append ' !platform:fedora'

File: scripts/setup_env [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                                  -r "${PROJECT_DIR}requirements.txt" \
+                                  -r "${PROJECT_DIR}test-requirements.txt" \
+                                  -r "${PROJECT_DIR}molecule-requirements.txt" ${@:-}
+}
+
+## Main ----------------------------------------------------------------------
+# Source distribution information
+source /etc/os-release || source /usr/lib/os-release
+RHT_PKG_MGR=$(command -v dnf || command -v yum)
+PYTHON_EXEC=$(command -v python3.10 || command -v python)
```
Comment (raukadah) [inline scripts/setup_env:41]: s/python3.10/python3 -> works in fedora 37

File: scripts/run-local-test [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## Vars ----------------------------------------------------------------------
+
+export PROJECT_DIR="$(dirname $(readlink -f ${BASH_SOURCE[0]}))/../"
+export ROLE_NAME="${ROLE_NAME:-$1}"
+
+# Ensure we don't keep any dangling sudo session in the shell
+# Run local test
+PS1="[\u@\h \W]\$" source "${HOME}/test-python/bin/activate"
+
+pushd "${PROJECT_DIR}ci_framework/roles/${ROLE_NAME}"
```
Comment (raukadah) [inline scripts/run-local-test:32]: on running make molecule
```
++ hash -r
+ pushd '/home/chandankumar/src/github.com/openstack-k8s-operators/ci-framework/scripts/../ci_framework/roles/*'
scripts/run-local-test: line 32: pushd: /home/chandankumar/src/github.com/openstack-k8s-operators/ci-framework/scripts/../ci_framework/roles/*: No such file or directory
make: *** [Makefile:14: molecule] Error 1

```

**Review by raukadah — COMMENTED** [review]
That's a great start. molecule tests work fine. I have left few comments.

File: scripts/run-local-test (hunk 3 above)
Comment (cjeanner) [reply scripts/run-local-test:32]: because there's no role yet as-is

File: bindep.txt (hunk 1 above)
Comment (cjeanner) [reply bindep.txt]: I removed that dependency for now - we don't use it afaik.

**Comment by sandeepyadav93** [comment]
lgtm thanks for adding `make help` and getting rid of TOX_CONSTRAINTS_FILE

I am hitting some issue running molecule tests but PR looks good to me in general.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+		bash scripts/run-local-test "$$(basename $${role})" ; \
+	done
+
+.PHONY: pre_commit
+pre_commit: setup_tests ## Runs pre-commit tests
+	${HOME}/test-python/bin/pip3 install pre-commit
+	${HOME}/test-python/bin/pre-commit run
+
+
+.PHONY: tests ## Run all tests
```
Comment (raukadah) [inline Makefile:27]: move "## Run all tests" to line 28
like this
```
tests: setup_tests pre_commit ## Run all tests
```
then make help will list tests also.

**Review by raukadah — CHANGES_REQUESTED** [review]
Just one comment, rest looks good. I will approve this patch after update.

**Review by raukadah — APPROVED** [review]

**Pull request merged**